client/lib/signup/step-actions/index.js

```diff
@@ -130,7 +130,7 @@ function getNewSiteParams( {
 
 	const forceAutoGeneratedBlogName = signupDependencies?.forceAutoGeneratedBlogName;
 	const shouldHideFreePlan = get( signupDependencies, 'shouldHideFreePlan', false );
-	const useAutoGeneratedBlogName = shouldHideFreePlan || forceAutoGeneratedBlogName;
+	const useAutoGeneratedBlogName = ! siteTitle || shouldHideFreePlan || forceAutoGeneratedBlogName;
```

Comment:
@alshakero now with skipping the domains step, we have `shouldHideFreePlan` false here.
I added a check for `siteTitle` to handle this new scenario. Skipping the domain selection much have autogenerated blog name since this is required by the backend